ksql-parser/src/main/java/io/confluent/ksql/parser/tree/AbstractStreamDropStatement.java

```diff
@@ -14,14 +14,65 @@
 
 package io.confluent.ksql.parser.tree;
 
+import com.google.common.base.MoreObjects;
+import java.util.Objects;
 import java.util.Optional;
 
 public abstract class AbstractStreamDropStatement extends Statement {
-  public AbstractStreamDropStatement(final Optional<NodeLocation> location) {
+
+  private final QualifiedName name;
+  private final boolean ifExists;
+  private final boolean deleteTopic;
+
+  public AbstractStreamDropStatement(final Optional<NodeLocation> location,
+                                     final QualifiedName name,
+                                     final boolean deleteTopic,
+                                     final boolean ifExists) {
     super(location);
+    this.name = name;
+    this.deleteTopic = deleteTopic;
+    this.ifExists = ifExists;
+  }
+
+  public QualifiedName getName() {
+    return name;
+  }
+
+  public boolean isDeleteTopic() {
+    return deleteTopic;
+  }
+
+  public boolean getIfExists() {
+    return ifExists;
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    final AbstractStreamDropStatement that = (AbstractStreamDropStatement) o;
```

Comment:
`equals` and `hashCode` aren't taking `super` state into account... is that intentional?

Reply:
Do we even need hashcode and equals? If we don't we should remove them. If we do, we should test them! (are you familiar with `EqualsTester`?)

Reply:
`super` doesn't actually have an implementation of either `hashCode`/`equals`. my guess is that we do actually need them since `Node` (the superclass) explicitly makes them `abstract` so that I need to implement it.

I am not familiar with `EqualsTester`! I'll look into it and hopefully add another tool to my Java arsenal 🔫

Reply:
I have a feeling you can remove these methods and everything will still compile and work.  There may be a few test failures, because the tests are relying on the _broken_ implementations, but maybe they would be better switched over to using a custom Hamcrest Matcher?

Reply:
`EqualsTester` rocks.   There's also one for NPE testing.